tools/parse_map_file.py

```diff
@@ -69,10 +91,18 @@ def read_gnu_map_file(file_name):
                             # Variable name and address found
                             address = int(match_results.groups()[0], 16)
                             variables[address] = match_results.groups()[1]
-    
+
     return (functions, variables)
 
 def main():
+    """Test retrieval of functions and variables dictionaries.
+
+    This module is not meant to be used directly when analyzing execution trace
```

Comment:
"Not meant to be used directly."

I don't know what is catching my eye here, but something is. Apologies for the gap from brain to action on my part. Bryan would be a good one to ask if this is "Pythonic", and how to refactor if it is not.

Reply:
It may not be Pythonic.  I am no Python expert.  My guess is that the Pythonic way would be that this is a package on PyPi.  As it is, these scripts work directly on the command line.  I suppose some long-term goal would have you adding this repo to your project using `FetchContent()` and installing the Python code into your venv using `pip install`.

Putting all of that aside, what this comment is trying to say is that `parse_map_file.py` is a submodule.  It will get invoked by the top level interface `trace_from_ser_port.py` or `trace_from_file.py`.  You _can_ however call this module directly, in which case it will spit out the functions and variables list directly to the terminal.